lua/tardis/sh_interiors.lua

```diff
@@ -76,47 +96,130 @@ function TARDIS:SaveIntPreferences(preferences)
     TARDIS:SetSetting("interior_preferences", preferences, true)
 end
 
-function TARDIS:IsFavoriteInt(id, ply)
-    return TARDIS:GetIntPreferences(ply).favorites[id]
+
+----------------------------------------------------------------------------------------------------
+-- Interior custom settings
+
+function TARDIS:GetMainVersionId(int_id)
+    return (self.Metadata[int_id] and self.Metadata[int_id].IsVersionOf) or int_id
 end
 
-function TARDIS:IsUnwantedInt(id, ply)
-    return TARDIS:GetIntPreferences(ply).unwanted[id]
+function TARDIS:GetIntCustomSetting(int_id, setting_id, ply, default_val)
+    local int_id = TARDIS:GetMainVersionId(int_id)
+    local metadata = self.Metadata[int_id]
+
+    local int_pref = TARDIS:GetIntPreferences(ply)
+    if not int_pref or not int_pref.custom_settings then return default_val end
+
+    local settings = int_pref.custom_settings[int_id]
+    if settings and settings[setting_id] ~= nil then
+        return settings[setting_id]
+    end
+
+    if setting_id == "preferred_spawn" then
+        return TARDIS:SetupPreferredSpawn(int_id, ply)
+    end
+
+    local md_settings = metadata.CustomSettings
+    if md_settings and md_settings[setting_id] and md_settings[setting_id].value ~= nil then
+        local metadata_value = md_settings[setting_id].value
+        TARDIS:SetIntCustomSetting(int_id, setting_id, metadata_value, ply)
```

Comment:
This is very confusing, why is a Get method calling the Set method?

Reply:
This is the default-value clause
If the setting is not found, it is being recieved from default value of metadata
If so, we save the value of the setting to be that default value

Reply:
Actually, I've double checked - it's not required any more
(was left there due to dozens of refactoring changes)
It's removed now